AB#4291 ReadMe Edits (#1542)
global-121/121-platform: README.md

  To prevent conflicts between projects or components using other versions of Node.js it is recommended to use a 'Node version manager'.  
  Make sure to install the version specified in the [`.node-version`](.node-version)-file
- Install Git (<https://git-scm.com/download/win>)
- Clone the Git repository: `git clone https://github.com/global-121/121-platform.git`
- Install Docker (<https://docs.docker.com/docker-for-windows/install/>)

### On Linux

#### Install dependencies
- Install Git: `sudo apt install git-all`
- Clone the Git repository: `git clone https://github.com/global-121/121-platform.git`
- Install Node.js: To install the required version of Node.js and to prevent conflicts between projects or components using other versions of Node.js, use [NVM - Node Version Manager](http://nvm.sh/).  

>   After installing NVM run: `nvm install-latest-npm`
> - `cd` to the directory of the cloned repository above and execute `nvm install`. In this way nvm "sees" which version of NodeJS needs to be installed.

Reviewer: ```suggestion
  After installing NVM run, from the root of this repository:  
  `nvm install && nvm install-latest-npm`
  So that the required version of Node.js and `npm` get installed.
```

### On macOS

#### Install dependencies
- Install Git: <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git> > Installing on macOS (section)
- Clone the Git repository: `git clone https://github.com/global-121/121-platform.git`
- Install Node.js  
  To install the required version of Node.js and to prevent conflicts between projects or components using other versions of Node.js, use [NVM - Node Version Manager](http://nvm.sh/).  
  After installing NVM run:

      nvm install && nvm install-latest-npm

> - `cd` to the directory of the cloned repository above and execute `nvm install`. In this way nvm "sees" which version of NodeJS needs to be installed.

Reviewer: See previous comment:

```
  After installing NVM run, from the root of this repository:  
  `nvm install && nvm install-latest-npm`
  So that the required version of Node.js and `npm` get installed.
```

## Setup Interfaces

> Install dependencies for the interfaces, from the root of this repository, run:

Reviewer: This feels like a step back.
I think it helps to "stay out of the specific parts and start-from-the-root" especially in this initial Getting-Started README.

> Install dependencies for the interfaces, from the individual interface directories (`interfaces/*`) run:

Reviewer: ```suggestion
Install all dependencies for all interfaces. From the root of this repository, run:
```

Author: @elwinschmitz  I wrote this because it was observed that when installing from root folder, repeatedly I have observed errors such as:
`Local package.json exists, but node_modules missing, did you mean to install?`

>     npm install

Reviewer: ```suggestion
    npm run install:interfaces
```

Or to install 1 specific interface's dependencies, run: (where `<interface-name>` is one of `pa`, `aw`, `ho`, `referral`)

    npm run install:<interface-name>

## Setup Services

> Install dependencies for the interfaces, from the individual interface directories (`services/*`) run:
> 
>     npm install
> 

Reviewer: This is only required if you want to work on a service in a 'standalone', local way.
It is not the most convenient way and needs some manual adjustments.

Services can/should run within the Docker environment by default. Any other way should only be done "when you know what you're doing".

```suggestion
```

Follow the "[Getting started / installation](services/README.md#getting-started--installation)"-section in the [services/README](services/README.md)-file.

> ## Start Interfaces
> To start all interfaces at once, execute `npm run start:interfaces` from the root of the project folder.

Reviewer: ```suggestion
After that, you can start in development mode by running from the root of this repository:
- `npm run start:services`

## Start Interfaces
To start all interfaces at once, execute `npm run start:interfaces` from the root of this repository.
```

  - Unit-tests and UI-tests for all interfaces; Run with `npm test` in each `interfaces/*`-folder.
  - Unit-tests and integration-tests for all services; Run with `npm test` in each `services/*`-folder.

## Common problems with Local Environment set-up
If the swagger UI is not accessible after installing docker and setting up services, we can take following steps to debug:

> 1. `docker ps` to list running containers and their statuses
> 2. `docker logs -f <container-name>` to check their statuses

Reviewer: ```suggestion
1. `docker-compose ps` to list running containers and their statuses
2. `docker-compose logs -f <container-name>` to check their statuses
```

We should use `docker-compose` as that is aware of all the other services in our `docker-compose.yml`-file, and not on the individual containers.

- If the errors are related to not being able to access/connect to 'database' then resetting the database by using `dropSchema: true` in `ormconfig.ts`.

> - In order to access docker container itself, execute command `sudo docker exec -it <container-name> sh` and verify installation of npm modules.

Reviewer: ```suggestion
- In order to access an individual docker container, execute command `docker-compose exec -it <container-name> sh` and verify installation of npm modules.
```

`sudo` shouldn't be necessary here. (Currently on our servers, it _is_; But that should be fixed there)

> ## Unit Tests

Reviewer: There is a section-header "Testing" at line 136: https://github.com/global-121/121-platform/pull/1542/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R136

I think that should be 'moved down here', so we get the order:

1. getting started / setup dev-environment
2. common issues (or as a sub-section of the previous)
3. testing
4. unit-tests (or as a sub-section, might be even better)
